### ISSUE TITLE
sv-dkl example timing out with pytorch-1.0

### ISSUE THREAD

**Issue by neerajprad** [pyro-ppl/pyro#1540]
I am seeing an increasing number of builds ([example](https://travis-ci.org/uber/pyro/jobs/455675436)) being aborted in `test_examples` with the error that `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.` on the `pytorch-1.0` branch. This seems to mostly get triggered in `python examples/contrib/gp/sv-dkl.py`. 

The problem usually resolves if we rebuild, but it is worth investigating as we are not seeing this on the `dev` branch. If the issue is due to #1421, we should make this test example faster so as not to stall the build.

```
python examples/contrib/gp/sv-dkl.py --epochs=1 --num-inducing=4
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

**Comment by fritzo** [comment]
I've seen a couple stalled builds on dev branch recently, so this may merely be a Travis issue. I'll grab details next time I see this.

**Comment by neerajprad** [comment]
I think the issue is most likely that the torchvision dataloader is deadlocking in pytorch-1.0 for some reason. I'm still investigating it.

**Comment by fehiepsi** [comment]
I have tried various assumptions in #1612 without successful. Here are my observations:

+ This seems like a DataLoader problem: I disable `forward` and `backward`, failed travis can be seen  at https://travis-ci.org/uber/pyro/jobs/465717303.
+ Because `vae` example also requires MNIST dataloader but it didn't fail in Travis. I thought that the torchvision Normalize transform might be the problem. However, the problem is still happened when I removed it. :(
+ The loss in failed run is different from the loss in passed run by a small margin. Loss is consistent in failed run and also consistent in passed run.

**Comment by neerajprad** [comment]
> The loss in failed run is different from the loss in passed run by a small margin. Loss is consistent in failed run and also consistent in passed run.

Maybe we can try `shuffle=False` and see if that at least gives the same loss.

**Comment by fehiepsi** [comment]
It seems that there is a regression in `conditional` function. In the commit which I mentioned above (about DataLoader problem), I forget to disable `forward` pass of `test` function. So DataLoader might not be a problem. I'm investigating more.

**Comment by fehiepsi** [comment]
Finally, I can isolate the slowness: `nn.Conv2d`.
+ When test passes ([line 1158](https://travis-ci.org/uber/pyro/jobs/466354602)): it took 0.00779986381531s.
+ When test fails ([line 1121](https://travis-ci.org/uber/pyro/jobs/466354604)): it took 0.7628870010375977s.

This [topic](https://github.com/pytorch/pytorch/issues/14456) might be relevant.

**Comment by neerajprad** [comment]
> Finally, I can isolate the slowness: nn.Conv2d.

Wow..great job debugging this! I wish this could have been fixed in release, because it seems like a serious regression (not so much for Pyro, but other users).

**Comment by fehiepsi** [comment]
@neerajprad It just happens on Travis. I can't replicate it locally even when I install PyTorch from pip: `https://download.pytorch.org/whl/cpu/torch-1.0.0-cp36-cp36m-linux_x86_64.whl` :(
